presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -233,6 +233,7 @@ private void processNewSources()
         long maxRuntime = duration.roundTo(TimeUnit.NANOSECONDS);
 
         Optional<ListenableFuture<?>> result = tryWithLock(100, TimeUnit.MILLISECONDS, () -> {
+            driverContext.getYieldSignal().setWithDelay(maxRuntime, driverContext.getYieldExecutor());
```

Comment:
I don't think it would be particularly useful because I don't expect the use of `setWithDelay` to be widespread. In fact, I expect this to be the only place (other than tests).